### PR TITLE
Tools: Add flag `--selector.relabel-config-file` / `--selector.relabel-config` / `--max-time` / `--min-time` to filter served blocks.

### LINKED ISSUE #4452: tool: bucket web add relabel and time flag
### Is your proposal related to a problem?

The bucket has many tsdb objects, the ui frontend will clunky or hang, it can not drag. I open chrome console to inspect, 
the response data size of api endpoint`/api/v1/blocks?view=global` is `12.4MB` .
image

### Describe the solution you'd like

add time filter and relabel config flag on `thanos tools bucket web` start like store component.

### PR DESCRIPTION
**Pull request by hanjm**
fix #4452 

Signed-off-by: hanjm <hanjinming@outlook.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

- add `min-time` `max-time` `selector.relabel-config` flag to `thanos tools bucket web`.

## Verification

e2e test.
run with new flag, it works.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 - [#3970](https://github.com/thanos-io/thanos/pull/3970) Azure: Adds more configuration options for Azure blob storage. This allows for pipeline and reader specific configuration. Implements HTTP transport configuration options. These options allows for more fine-grained control on timeouts and retries. Implements MSI authentication as second method of authentication via a service principal token.
 - [#4406](https://github.com/thanos-io/thanos/pull/4406) Tools: Add retention command for applying retention policy on the bucket.
 - [#4430](https://github.com/thanos-io/thanos/pull/4430) Compact: Add flag `downsample.concurrency` to specify the concurrency of downsampling blocks.
+- [#4453](https://github.com/thanos-io/thanos/pull/4453) Tools: Add flag `--selector.relabel-config-file` / `--selector.relabel-config` / `--max-time` / `--min-time` to filter served blocks.
```
Comment (yeya24) [inline CHANGELOG.md:33]: As 0.22.0 rc is there, let's add this to the `Unreleased` section? WDYT @hanjm

**Review by yeya24 — DISMISSED** [review]
LGTM with only one nit. Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (hanjm) [reply CHANGELOG.md]: You are right. Done, Thanks.

**Comment by hanjm** [comment]
I have add e2e test, @yeya24 could you help me review again? thank you.

File: test/e2e/tools_bucket_web_test.go [hunk 2]
```diff
@@ -4,11 +4,24 @@
...
 	"testing"
+	"time"
 
 	"github.com/cortexproject/cortex/integration/e2e"
 	e2edb "github.com/cortexproject/cortex/integration/e2e/db"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	v1 "github.com/thanos-io/thanos/pkg/api/blocks"
+	"github.com/thanos-io/thanos/pkg/objstore"
```
Comment (yeya24) [inline test/e2e/tools_bucket_web_test.go:24]: Let format the imports. Please move thanos to the group below.

File: test/e2e/e2ethanos/services.go [hunk 3]
```diff
@@ -662,7 +662,14 @@ func NewMemcached(name string) *e2e.ConcreteService {
...
 }
 
-func NewToolsBucketWeb(name string, bucketConfig client.BucketConfig, routePrefix, externalPrefix string) (*Service, error) {
+func NewToolsBucketWeb(
+	name string,
+	bucketConfig client.BucketConfig,
+	routePrefix,
+	externalPrefix string,
+	minTime string,
+	maxTime string,
```
Comment (yeya24) [inline test/e2e/e2ethanos/services.go:671]: I feel using time.Time is better than string. We can convert them to RFC3339 format below

**Review by yeya24 — DISMISSED** [review]
Excellent work! Just two small nits.

File: test/e2e/tools_bucket_web_test.go (hunk 2 above)
Comment (hanjm) [reply test/e2e/tools_bucket_web_test.go:25]: Fixed.

File: test/e2e/e2ethanos/services.go (hunk 3 above)
Comment (hanjm) [reply test/e2e/e2ethanos/services.go:671]: but it it the [pkg/model/TimeOrDurationValue type](https://github.com/thanos-io/thanos/blob/main/pkg/model/timeduration.go#L14), could be absolute time or relevant time duration, but the string value `(*TimeOrDurationValue.)String()` can not be parsed at `(*TimeOrDurationValue.).Set(string)`, so just use string for simply.

Comment (yeya24) [reply test/e2e/e2ethanos/services.go:671]: That makes sense. 💪

**Review by yeya24 — APPROVED** [review]
lgtm

**Pull request merged**